app/controllers/admin/quizzs_controller.rb

```diff
@@ -0,0 +1,116 @@
+class Admin::QuizzsController < ApplicationController
+
+  before_action :authenticate_user!
+  before_action :admin_action?
+
+  before_action :set_quizz, only: [:show, :edit, :update, :destroy, :add_question, :save_question, :complete]
+  
+  before_action :set_quizzes, only: [:index, :update, :save_question]
+
+  respond_to :js, only: [:new, :edit, :create, :update, :add_question]
```

Comment:
questions should be moved to separate controller as nested routes, with own CRUD